### PR TITLE
use codecov integration

### PR DESCRIPTION
**Pull request by MartinNowak**
- patch coverage is highly useful

Still needs to be enabled under https://github.com/apps/codecov, might need to ask Andrei for that.

### CODE REVIEW

**Comment by wilzbach** [comment]
> might need to ask Andrei for that

CC @andralex (dlang/tools would be another good candidate)

File: .codecov.yml [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
-  # range: "70...100"
+  round: down
+  range: "80...100"
 
   status:
     # Learn more at https://docs.codecov.io/docs/commit-status
     project: true
     patch: true
-    changes: false
+    changes: true
```
Comment (wilzbach) [inline .codecov.yml]: Not sure about the effects/usefulness of this one, but I guess it doesn't hurt to try.

File: .codecov.yml [hunk 2]
```diff
@@ -1,17 +1,14 @@
 # Documentation: https://docs.codecov.io/docs/codecov-yaml
 
-codecov:
-  bot: dlang-bot
-
```
Comment (wilzbach) [inline .codecov.yml]:
```
codecov:
  ci:
  notify:
    after_n_builds: 1
```

Seems to be a sensible default as well.

**Comment by andralex** [comment]
noice...

**Comment by andralex** [comment]
Added dub to codecov.

File: .codecov.yml (hunk 1 above)
Comment (MartinNowak) [inline .codecov.yml]: Let's check, a patch shouldn't reduce coverage in previously covered areas.

File: .codecov.yml (hunk 2 above)
Comment (MartinNowak) [inline .codecov.yml]: Not sure it's seems sensible to me to enable an advanced configuration by default?
Don't they know how to deal with CI? I mean eventually they should just react to us sending sth. from a CI run.

Comment (wilzbach) [inline .codecov.yml]: Yes eventually they will, but by default they will wait for all CIs. Those four lines will trigger the notification on the first submission - don't ask me why they don't send it by default.

**Pull request merged**